### DISCUSSION TITLE
How to insert an image outside the visible page rectangle

### DISCUSSION THREAD

**Discussion by unstoppableking** [pymupdf/PyMuPDF#2780] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
the coordinate of fitz.rect make some mistakes in funtion of  page.insert_image()

## To Reproduce (mandatory)
page.rect is Rect(0.0, 0.0, 2384.0, 1684.0)
img_rect = fitz.Rect([2133, 1002, 2351, 1084])
page.insert_image(img_rect, filename=stamp_path[1], overlay = False,  rotate = 0)

For problems when building or installing PyMuPDF, give the full output of the build/install command so that, for example, all pip/compiler/linker errors/warnings can be seen.

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).
insert image in the rectangle of [2133, 1002, 2351, 1084], like watermark

## Screenshots (optional)
![image](https://github.com/pymupdf/PyMuPDF/assets/49112111/67f724d6-0ce6-406c-9944-449a5652441d)
![image](https://github.com/pymupdf/PyMuPDF/assets/49112111/28daddc1-8c25-4681-8429-63a76847c258)

## Your configuration (mandatory)
-system:window 10
 - Python 3.8.17 (default, Jul  5 2023, 20:44:21) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
 -PyMuPDFb==1.23.5
For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

**Comment by JorjMcKie** [comment]
This is normal and no issue at all. Please consult the documentation: The pag.rect is the **visible** **part** of the page - necessarily the full MediaBox.

**Reply by unstoppableking** [reply]
So how to put the rect in the position which is out of visible part of the page? 
I tried create new page, and my location of rect is right. but it didn't work in old page.

**Reply by unstoppableking** [reply]
maybe I should retell my problem.
This picture is what I want (It's in visible part of page, producted by create a new page)
![image](https://github.com/pymupdf/PyMuPDF/assets/49112111/aa81ff16-3f55-4c64-803a-c448e71621de)

And this picture is the result of inserting image in old page(It's also in visible part of page, producted by page.insert_image)
![image](https://github.com/pymupdf/PyMuPDF/assets/49112111/a7c41c64-5ab1-4460-b182-d54e9a3788a6)

Here is my code:
def stamping(page, stamp_position, stamp_path, page_size):
    doc = fitz.open()
    page1 = doc.new_page(width = page.rect.width,
                           height = page.rect.height)
    coord_transform1 = [int(i/0.352777777778) for i in stamp_position[0]]
    coord_transform2 = [int(i/0.352777777778) for i in stamp_position[1]]
    img_rect1 = fitz.Rect(coord_transform1[0], coord_transform1[1], coord_transform1[2],  coord_transform1[3])
    coord_transform2 = [i/0.352777777778 for i in stamp_position[1]]
    img_rect2 = fitz.Rect(coord_transform2[0], coord_transform2[1], coord_transform2[2],  coord_transform2[3])
    
    img1 = fitz.open(stamp_path[0]) 
    pdfbytes1 = img1.convert_to_pdf()
    img1.close()
    imgPDF1 = fitz.open("pdf", pdfbytes1)
    img2 = fitz.open(stamp_path[1]) 
    pdfbytes2 = img2.convert_to_pdf()
    img2.close()
    imgPDF2 = fitz.open("pdf", pdfbytes2)
    page1.show_pdf_page(img_rect1, imgPDF1, 0)
    page1.show_pdf_page(img_rect2, imgPDF2, 0)
    
    page.draw_rect(img_rect1)
    page.draw_rect(img_rect2)
    page.insert_image(img_rect1, filename=stamp_path[0], overlay = False, rotate = 270)
    page.insert_image(img_rect2, filename=stamp_path[1], overlay = False,  rotate = 270)

    doc.save('f%s.pdf' %page_size[0])
    
    return page

**Comment by JorjMcKie** [comment, marked as answer]
You can set the page's visible part to the full "physical" page, then insert what you want, then restore to the original page.

Roughly do this (assuming page rotation = 0 for simplifying the explanation):
```python
cropbox = page.cropbox  # save the old CropBox value
page.set_cropbox(page.mediabox)  # set visible page equal to full page
page.insert....  # insert what you need
# optionally re-instate the old visible part
page.set_cropbox = cropbox
```

**Reply by unstoppableking** [reply]
That is actually what I need! No words can adequately express my appreciation : )

**Reply by JorjMcKie** [reply]
I am pleased this did help you!
